src/test/java/uk/gov/pay/webhooks/deliveryqueue/managed/WebhookMessagePollingServiceTest.java

```diff
@@ -0,0 +1,138 @@
+            .addEntityClass(EventTypeEntity.class)
+            .addEntityClass(WebhookMessageEntity.class)
+            .addEntityClass(WebhookEntity.class)
+            .addEntityClass(WebhookDeliveryQueueEntity.class)
+            .build();
+
+    private InstantSource instantSource;
+    private WebhookDeliveryQueueDao webhookDeliveryQueueDao;
+    private WebhookMessageDao webhookMessageDao;
+    private WebhookDao webhookDao;
+    private WebhookMessagePollingService webhookMessagePollingService;
+    private WebhookMessageSender webhookMessageSenderMock;
+    private SendAttempter sendAttempter;
+    @Mock
+    private HttpResponse response;
+
+    @BeforeEach
+    public void setUp() throws IOException, InvalidKeyException, InterruptedException {
+        var environment = mock(Environment.class);
+
+        when(environment.metrics()).thenReturn(mock(MetricRegistry.class));
+        when(response.statusCode()).thenReturn(200);
+        webhookMessageSenderMock = mock(WebhookMessageSender.class);
+        when(webhookMessageSenderMock.sendWebhookMessage(any())).thenReturn(response);
+        instantSource = InstantSource.fixed(Instant.now());
+        webhookMessageDao = new WebhookMessageDao(database.getSessionFactory());
+        webhookDao = new WebhookDao(database.getSessionFactory());
+        webhookDeliveryQueueDao = new WebhookDeliveryQueueDao(database.getSessionFactory(), instantSource);
+
+        sendAttempter = new SendAttempter(webhookDeliveryQueueDao, instantSource, webhookMessageSenderMock, environment);
+        webhookMessagePollingService = new WebhookMessagePollingService(webhookDeliveryQueueDao, sendAttempter, database.getSessionFactory());
+    }
+
+    @Test
+    public void shouldPollWithNoActionsOnEmptyQueue() {
+       webhookMessagePollingService.pollWebhookMessageQueue();
+       verifyNoInteractions(webhookMessageSenderMock);
+    }
+
+    @Test
+    public void shouldEmitSingleValidQueueItem() throws IOException, InvalidKeyException, InterruptedException {
+        setupOrderedDeliveryQueueWith(List.of("first-external-id"));
+        webhookMessagePollingService.pollWebhookMessageQueue();
+        verify(webhookMessageSenderMock).sendWebhookMessage(argThat((webhookMessage -> webhookMessage.getExternalId().equals("first-external-id"))));
+    }
+
+    @Test
+    public void shouldEmitMultipleValidQueueItemsInOnePoll() throws IOException, InvalidKeyException, InterruptedException {
+        setupOrderedDeliveryQueueWith(List.of("first-external-id", "second-external-id"));
+        webhookMessagePollingService.pollWebhookMessageQueue();
+
+        var captor = ArgumentCaptor.forClass(WebhookMessageEntity.class);
+        verify(webhookMessageSenderMock, times(2)).sendWebhookMessage(captor.capture());
+        assertEquals("first-external-id", captor.getAllValues().get(0).getExternalId());
+        assertEquals("second-external-id", captor.getAllValues().get(1).getExternalId());
```

Comment:
I suspect in-order verification with `argThat(…)` might work here (not tat there’s anything wrong with this).